Added file fetcher

aklyachkin opened the pull request:
Signed-off-by: aklyachkin <aklyachkin@gmail.com>

### Description

Added file fetcher to copy just a file (not the whole path).

--------------------------------------------------
/cc @chef/omnibus-maintainers <- This ensures the Omnibus Maintainers team are notified to review this PR.

#### Maintainers

Please ensure that you check for:

- [] If this change impacts git cache validity, it bumps the git cache
  serial number
- [] If this change impacts compatibility with omnibus-software, the
  corresponding change is reviewed and there is a release plan
- [] If this change impacts compatibility with the omnibus cookbook, the
  corresponding change is reviewed and there is a release plan

srenatus commented on lib/omnibus/fetchers/file_fetcher.rb:
```diff
@@ -0,0 +1,131 @@
+    #
+    # The shasum of the directory **inside** the project.
+    #
+    # @return [String]
```
[nit] This isn't correct, is it? I think I'd rather remove [line 119](https://github.com/chef/omnibus/pull/795/files#diff-33e4fde83dc6d509149df509f68bd6eeR119) and make this spec `@return [String, nil]` 🤔

srenatus commented on lib/omnibus/fetchers/file_fetcher.rb:
```diff
@@ -0,0 +1,131 @@
+    end
+
+    #
+    # The shasum of the directory **inside** the project.
```
[nit] this still says "directory"

srenatus commented on lib/omnibus/fetchers/file_fetcher.rb:
```diff
@@ -0,0 +1,131 @@
+    end
+
+    #
+    # The shasum of the directory **outside** of the project.
```
[nit] "directory"

aklyachkin commented:
Thanks for the review!  I've copied path fetcher and changed it, but it seems, I forgot change some wordings :-) I did it know. And you're right it is nonsense to return 0 in the function, which should return "String". I changed it as you told.
